src/Controls/src/Core/Element.cs

```diff
@@ -118,13 +211,10 @@ internal IEnumerable<Element> AllChildren
 		// return null by default so we don't need to foreach over an empty collection in OnPropertyChanged
 		internal virtual IEnumerable<Element> ChildrenNotDrawnByThisElement => null;
 
-		IReadOnlyList<Element> IElementController.LogicalChildren => LogicalChildrenInternal;
+		IReadOnlyList<Element> _logicalChildrenInternal;
+		IList<Element> _internalChildren;
 
-		/// <include file="../../docs/Microsoft.Maui.Controls/Element.xml" path="//Member[@MemberName='LogicalChildren']/Docs/*" />
-		[EditorBrowsable(EditorBrowsableState.Never)]
-		[Obsolete("Do not use! This is to be removed! Just used by Hot Reload! To be replaced with IVisualTreeElement!")]
```

Comment:
@drasticactions is this actually still used by Hot Reload? If I delete this, will it break anything? Or have you all switched over to purely using IVTE?

Reply:
All MAUI Hot Reload uses IVisualTreeElement, if it's not there you won't see it in the Live Visual Tree in VS (And so, not tracked for Hot Reload). We don't directly reference internal children elements anymore, just IVTE.

Reply:
Hmm it seems like something is still using it?

<img width="840" alt="image" src="https://user-images.githubusercontent.com/5375137/230523622-3220f22b-2563-4644-b5a7-72a36d978fb1.png">

<img width="1002" alt="image" src="https://user-images.githubusercontent.com/5375137/230524938-560768b4-f407-41b1-bc40-2787ac30af44.png">

Reply:
Well, I lied, lol

https://github.com/xamarin/UITools/blob/d33449e6488f8605916a1166b97423652a1ed2e8/HotReload/Source/Xamarin.HotReload.Xaml/VisualTreeService/VisualTree/FormsVisualDiagnostics.cs#L113-L128

https://github.com/xamarin/UITools/blob/d33449e6488f8605916a1166b97423652a1ed2e8/HotReload/Source/Xamarin.HotReload.Xaml/VisualTreeService/VisualTree/FormsVisualDiagnostics.cs#L82-L88

That was a hack to avoid a Forms issue where we were getting -1 indexes. That _should_ be okay for MAUI so it should be safe to remove it, I think.

Reply:
> That was a hack to avoid a Forms issue where we were getting -1 indexes. That should be okay for MAUI so it should be safe to remove it, I think. (as in, safe for us to remove the hack and rely on the actual value. If we get -1s, those should be logged so we can fix it in MAUI)

Aw ok cool, look at that. It was being hit because the ordering on window for adding logical children was wrong.

Reply:
Removing the workaround, https://github.com/xamarin/UITools/pull/1531

Reply:
If we remove this api, we will require an IDE update all around. Maybe we keep it for now and remove once we have reached 90% IDE Update?

Reply:
@mattleibow Maybe it could be no-oped? (Ex. Create an empty list and have it always return that. Hot Reload won't work, but your app won't blow up), or it could return whatever the new API is and have an obsolete tag around it directing those to the new API?

Reply:
Yeah, rather have hot reload work.

Reply:
The UITools code try/catches the call to `LogicalChildren` so do we care? At runtime won't it still just return a -1, log the exception, and then Hot Reload continues to work? 

https://github.com/xamarin/UITools/blob/main/HotReload/Source/Xamarin.HotReload.Xaml/VisualTreeService/VisualTree/FormsVisualDiagnostics.cs#L129-L134

Reply:
So if the ChildIndex is -1, XET (AFAIK, @etvorun could correct me) then we need would reinit the nodes of the parent that called it. So, in this case, we would need to do that for every element in the chain, since it would always return -1. Hot Reload would still "work" but it would, most likely, have different behavior than what we normally have (for better or worse) and the performance would probably take a hit.

Also, I wrote a Debug.WriteLine (ex); so that would spam the output. I should change that to go back to the proper IDE channels instead so it can get logged correctly, but that would also spam the output until we remove that code.

As an aside, this could also create issues with versioning: While we can tie this release of hot reload to this release of MAUI with this code, VS still would support net6.0 and 7.0 MAUI, etc. Removing the workaround would cause all versions to use the ChildIndex value from MAUI, and I'm not sure if that could cause issues or not.

@etvorun What do you think? Do you have any thoughts?

Reply:
Add the obsolete LogicalChildren back for now

Reply:
Nice!